src/pybind/mgr/dashboard/services/exception.py

```diff
@@ -51,6 +51,9 @@ def dashboard_exception_handler(handler, *args, **kwargs):
         cherrypy.response.headers['Content-Type'] = 'application/json'
         cherrypy.response.status = getattr(e, 'status', 400)
         return json.dumps(serialize_dashboard_exception(e)).encode('utf-8')
+    except Exception as error:
+        logger.exception('Internal Server Error')
+        raise error
```

Comment:
Shan't we return the error?

```suggestion
    except Exception as error:
        logger.exception('Internal Server Error')
        cherrypy.response.headers['Content-Type'] = 'application/json'
        cherrypy.response.status = getattr(e, 'status', 500)
        raise error
```

Reply:
@epuertat It depends on what we consider appropriate: if the API consumer should be informed about the details of this kind of internal errors, then yes we can do it (in that case I wouldn't re-raise the error).

Reply:
I vote to get as much as possible information because i'm always asking myself what exactly happens when this error pops up as notification in the dashboard. I want to see WHAT happens, not only a simple text without useful information.

Reply:
I agree, it would be useful to get some insight into why it crashed. Maybe hidden behind a "Details" button, to avoid cluttering/flooding the UI too much?

Reply:
> @epuertat It depends on what we consider appropriate: if the API consumer should be informed about the details of this kind of internal errors, then yes we can do it (as we're adding more info, I wouldn't re-raise the error).
> 
> Currently, the user receives a response like this:
> 
> ```
> {"status": "500 Internal Server Error", "detail": "The server encountered an unexpected condition which prevented it from fulfilling the request.", "request_id": "905d4bbb-7f6b-4a37-905d-26b5b610cbfd"}
> ```

With debug enabled, we should display as much info as possible in requests. It could be the case that the cherrypy response distpatch is itself triggering the exception (as I think it's happening to @fabrizio8, isn't it?), but the for every other internal error, we would get a traceback.

Reply:
> I agree, it would be useful to get some insight into why it crashed. Maybe hidden behind a "Details" button, to avoid cluttering/flooding the UI too much?

@LenzGr @epuertat @votdev I can investigate if we can show only the error message in the toasties,
but when you open the notification sidebar you get the full traceback.
Thoughts?

Reply:
> @LenzGr @epuertat @votdev I can investigate if we can show only the error message in the toasties,
> but when you open the notification sidebar you get the full traceback.
> Thoughts?

Sounds good, thank you! @tspmelo might be able to give you some hints on how to accomplish this.

Reply:
> @LenzGr @epuertat @votdev I can investigate if we can show only the error message in the toasties,
but when you open the notification sidebar you get the full traceback.
Thoughts?

I also prefer to not display it in the toasty. 
It will require some modifications, but it's possible to save that information and display it in the sidebar.

Since the error messages are going to be long, maybe it will be better to, instead of showing it directly in the sidebar, just insert a button in the notification that opens a modal with all the information.

Reply:
@LenzGr @tspmelo @LenzGr I created a separate issue for UI enhancements:
https://tracker.ceph.com/issues/46909

This PR ensures that error traceback is logged regardless debug mode enabled or disabled.